### PR TITLE
Add functional tests for dooh support 

### PR DESCRIPTION
**Pull request by osulzhenko**

### CODE REVIEW

**Review by marki1an — CHANGES_REQUESTED** [review]
Also missing tests for those requirements:
**Allow bidders to specify in their configuration if they support media types for a new dooh capability.**

File: src/test/groovy/org/prebid/server/functional/model/request/auction/DoohExt.groovy [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package org.prebid.server.functional.model.request.auction
+
+import groovy.transform.EqualsAndHashCode
+import groovy.transform.ToString
+
+@ToString(includeNames = true, ignoreNulls = true)
+@EqualsAndHashCode
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/model/request/auction/DoohExt.groovy:7]: Do we need here `@EqualsAndHashCode`?

File: src/test/groovy/org/prebid/server/functional/model/request/auction/DoohExtData.groovy [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package org.prebid.server.functional.model.request.auction
+
+import groovy.transform.EqualsAndHashCode
+import groovy.transform.ToString
+import org.prebid.server.functional.util.PBSUtils
+
+@ToString(includeNames = true, ignoreNulls = true)
+@EqualsAndHashCode
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/model/request/auction/DoohExtData.groovy:8]: Same here

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy [hunk 3]
```diff
@@ -36,46 +39,43 @@ class BidValidationSpec extends BaseSpec {
...
+        bidRequest << [BidRequest.defaultBidRequest.tap { it.app = App.defaultApp },
+                       BidRequest.defaultBidRequest.tap { it.dooh = Dooh.defaultDooh },
+                       BidRequest.getDefaultBidRequest(DistributionChannel.APP).tap {
+                           it.dooh = Dooh.defaultDooh
+                       },
+                       BidRequest.defaultBidRequest.tap {
+                           it.dooh = Dooh.defaultDooh
+                           it.app = App.defaultApp
+                       }
+        ]
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:61]:
```suggestion
 BidRequest.getDefaultBidRequest(DistributionChannel.APP).tap {
                           it.dooh = Dooh.defaultDooh
                       },
 BidRequest.getDefaultBidRequest(DistributionChannel.SITE).tap {
                           it.dooh = Dooh.defaultDooh
                       },
 BidRequest.getDefaultBidRequest(DistributionChannel.SITE).tap {
                           it.app = App.defaultApp
```

File: src/test/groovy/org/prebid/server/functional/tests/BidderParamsSpec.groovy [hunk 4]
```diff
@@ -510,6 +514,59 @@ class BidderParamsSpec extends BaseSpec {
...
+        assert !bidResponse.seatbid
+
+        and: "Should not send any bidRequest"
+        assert !bidder.getBidderRequests(bidRequest.id)
+
+        where:
+        mediaType                       | bidRequest
+        VIDEO.value                     | BidRequest.getDefaultBidRequest(DOOH)
+        NATIVE.value                    | BidRequest.getDefaultBidRequest(DOOH)
+        AUDIO.value                    | BidRequest.getDefaultBidRequest(DOOH)
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/BidderParamsSpec.groovy:539]: Formant

File: src/test/groovy/org/prebid/server/functional/tests/BidderParamsSpec.groovy [hunk 5]
```diff
@@ -510,6 +514,59 @@ class BidderParamsSpec extends BaseSpec {
...
+        when: "Requesting PBS auction"
+        def bidResponse = pbsService.sendAuctionRequest(bidRequest)
+
+        then: "BidderResponse should contain proper warning"
+        assert bidResponse.ext?.warnings[ErrorType.GENERIC]?.message.contains("Bid request contains 0 impressions after filtering.")
+
+        and: "BidderResponse should not contain any seatbid"
+        assert !bidResponse.seatbid
+
+        and: "Should not send any bidRequest"
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/BidderParamsSpec.groovy:532]: Should't send any bidder request

File: src/test/groovy/org/prebid/server/functional/tests/BidderParamsSpec.groovy [hunk 6]
```diff
@@ -510,6 +514,59 @@ class BidderParamsSpec extends BaseSpec {
...
+                 "adapters.generic.meta-info.dooh-media-types": mediaType])
+
+        when: "Requesting PBS auction"
+        def bidResponse = pbsService.sendAuctionRequest(bidRequest)
+
+        then: "BidderResponse should contain proper warning"
+        assert bidResponse.ext?.warnings[ErrorType.GENERIC]?.message.contains("Bid request contains 0 impressions after filtering.")
+
+        and: "BidderResponse should not contain any seatbid"
+        assert !bidResponse.seatbid
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/BidderParamsSpec.groovy:530]: Bid response shouldn't contain any seatbid(Not a bidder response)

File: src/test/groovy/org/prebid/server/functional/tests/BidderParamsSpec.groovy [hunk 7]
```diff
@@ -510,6 +514,59 @@ class BidderParamsSpec extends BaseSpec {
...
+        given: "PBS service with configuration for dooh media-types"
+        def pbsService = pbsServiceFactory.getService(
+                ["auction.filter-imp-media-type.enabled"      : "true",
+                 "adapters.generic.meta-info.dooh-media-types": mediaType])
+
+        when: "Requesting PBS auction"
+        def bidResponse = pbsService.sendAuctionRequest(bidRequest)
+
+        then: "BidderResponse should contain proper warning"
+        assert bidResponse.ext?.warnings[ErrorType.GENERIC]?.message.contains("Bid request contains 0 impressions after filtering.")
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/BidderParamsSpec.groovy:527]: Same here:
Bid response should contain proper warning(Not a bidder response)

File: src/test/groovy/org/prebid/server/functional/tests/BidderParamsSpec.groovy (hunk 6 above)
Comment (marki1an) [reply src/test/groovy/org/prebid/server/functional/tests/BidderParamsSpec.groovy:530]: And please check other places

**Review by marki1an — CHANGES_REQUESTED** [review]

File: src/test/groovy/org/prebid/server/functional/model/request/auction/DoohExtData.groovy (hunk 2 above)
Comment (osulzhenko) [reply src/test/groovy/org/prebid/server/functional/model/request/auction/DoohExtData.groovy:8]: Without it, test case: `PBS shouldn't remove bidRequest.dooh when PBS support ortb 2.6` will drop

**Review by marki1an — APPROVED** [review]

**Pull request merged**